### ISSUE TITLE
EAPTLS source doesn't seem to work with Eduroam config?

### ISSUE THREAD

**Issue by nathanbloemendaal78** [inverse-inc/packetfence#7828]
**Describe the bug**
I suspect it is a bug, not 100% sure, but here I go:
PF12.2 setup with latest patches. One AD-source, one EAP-TLS source (for device/machine authentication of SCEP delivered certificates using SCEPman in Azure. And an Eduroam exclusive source.

Authentication of local AD-users works fine, I defined the realm in use by AD as local realm. Authentication of external users on site also work fine via Eduroam. Internal users on external Eduroam-site also works fine.

One thing doesn't work, and that is the EAP-TLS source. It simply doesn't get hit. The device certificates in use have no suffix, so no @ in the username/CN/SAN. Therefore, I also defined the NULL source as local source in the Eduroam source. What does happen, is PF validates the certificate, outputs an OK and the device in question is put in the VLAN pool on the wireless controller (last resort):

610809) Mon Sep 25 10:37:12 2023: Debug:   } # authorize = updated
(610809) Mon Sep 25 10:37:12 2023: WARNING: You set Proxy-To-Realm = default, but it is a LOCAL realm!  Cancelling proxy request.
(610809) Mon Sep 25 10:37:12 2023: Debug: Found Auth-Type = eaptls-profile
(610809) Mon Sep 25 10:37:12 2023: Debug: # Executing group from file /usr/local/pf/raddb/sites-enabled/eduroam
(610809) Mon Sep 25 10:37:12 2023: Debug:   Auth-Type cust1-eap-tls-scepman {
(610809) Mon Sep 25 10:37:12 2023: Debug: eaptls-profile: Expiring EAP session with state 0xb972594dbca95480
(610809) Mon Sep 25 10:37:12 2023: Debug: eaptls-profile: Finished EAP session with state 0xac59d4ceab86d990
(610809) Mon Sep 25 10:37:12 2023: Debug: eaptls-profile: Previous EAP request found for state 0xac59d4ceab86d990, released from the list
(610809) Mon Sep 25 10:37:12 2023: Debug: eaptls-profile: Peer sent packet with method EAP TLS (13)
(610809) Mon Sep 25 10:37:12 2023: Debug: eaptls-profile: Calling submodule eap_tls to process data
(610809) Mon Sep 25 10:37:12 2023: Debug: eap_tls: (TLS) Peer ACKed our handshake fragment.  handshake is finished
(610809) Mon Sep 25 10:37:12 2023: Debug: eaptls-profile: Sending EAP Success (code 3) ID 223 length 4
(610809) Mon Sep 25 10:37:12 2023: Debug: eaptls-profile: Freeing handler
(610809) Mon Sep 25 10:37:12 2023: Debug:     [eaptls-profile] = ok
(610809) Mon Sep 25 10:37:12 2023: Debug:   } # Auth-Type eaptls-profile = ok
(610809) Mon Sep 25 10:37:12 2023: Debug: # Executing section post-auth from file /usr/local/pf/raddb/sites-enabled/eduroam
(610809) Mon Sep 25 10:37:12 2023: Debug:   post-auth {
(610809) Mon Sep 25 10:37:12 2023: Debug:     update {
(610809) Mon Sep 25 10:37:12 2023: Debug:       EXPAND %{Packet-Src-IP-Address}
(610809) Mon Sep 25 10:37:12 2023: Debug:          --> 10.7.1.101
(610809) Mon Sep 25 10:37:12 2023: Debug:     } # update = noop
(610809) Mon Sep 25 10:37:12 2023: Debug:     if (&control:Realm == "eduroam") {
(610809) Mon Sep 25 10:37:12 2023: Debug:     if (&control:Realm == "eduroam")  -> FALSE
(610809) Mon Sep 25 10:37:12 2023: Debug: attr_filter.packetfence_post_auth: EXPAND %{User-Name}
(610809) Mon Sep 25 10:37:12 2023: Debug: attr_filter.packetfence_post_auth:    --> host/hostname-shared
(610809) Mon Sep 25 10:37:12 2023: Debug: attr_filter.packetfence_post_auth: Matched entry DEFAULT at line 10
(610809) Mon Sep 25 10:37:12 2023: Debug:     [attr_filter.packetfence_post_auth] = updated
(610809) Mon Sep 25 10:37:12 2023: Debug: linelog: EXPAND messages.%{%{reply:Packet-Type}:-default}
(610809) Mon Sep 25 10:37:12 2023: Debug: linelog:    --> messages.Access-Accept
(610809) Mon Sep 25 10:37:12 2023: Debug: linelog: EXPAND [mac:%{Calling-Station-Id}] Accepted user: %{reply:User-Name} and returned VLAN %{reply:Tunnel-Private-Group-ID}
(610809) Mon Sep 25 10:37:12 2023: Debug: linelog:    --> [mac:xx:xx:x:xx:xx:xx] Accepted user: host/hostname-shared and returned VLAN
(610809) Mon Sep 25 10:37:12 2023: Debug: linelog: EXPAND stdout
(610809) Mon Sep 25 10:37:12 2023: Debug: linelog:    --> stdout
(610809) Mon Sep 25 10:37:12 2023: Debug:     [linelog] = ok
(610809) Mon Sep 25 10:37:12 2023: Debug:   } # post-auth = updated
(610809) Mon Sep 25 10:37:12 2023: Debug: Sent Access-Accept Id 104 from 10.x.x.x:11812 to 10.x.x.x:59972 length 188
(610809) Mon Sep 25 10:37:12 2023: Debug:   MS-MPPE-Recv-Key = 0x5509a279781922671eea27053668a99e56d24767a01972aa8952e9e7f2df507b
(610809) Mon Sep 25 10:37:12 2023: Debug:   MS-MPPE-Send-Key = 0xf8debe83b3fd80873ac342f06b1163b5f794a27581b26d26fd24ec4cb3d3022b
(610809) Mon Sep 25 10:37:12 2023: Debug:   EAP-Message = 0x03df0004
(610809) Mon Sep 25 10:37:12 2023: Debug:   Message-Authenticator = 0x00000000000000000000000000000000
(610809) Mon Sep 25 10:37:12 2023: Debug:   User-Name = "host/hostname-shared"
(610809) Mon Sep 25 10:37:12 2023: Debug: Finished request

So, it uses the eap tls profile the certificate, confirms it is OK and does not do the authorization step. The authentication does not show up on the audit logs, nor is it visible in packetfence.log. Thanks to the last resort VLAN pool on the wireless controller, it gets put in a VLAN, but not using the EAP-TLS source. I linked the EAP-TLS source to the NULL realm.
Other setups I have without Eduroam do not show this behavior and there, the EAP-TLS source is being used as expected, also with certificates without a realm.

**Expected behavior**
I would expect the EAPTLS source to be used just as the AD source.

Can provide config files if needed.

**Comment by nathanbloemendaal78** [comment]
Not even a short reply after 5 months?

**Comment by rubeste** [comment]
Hello. Do you still have this issue? If so could you please show the config of your connection profile?

**Comment by nathanbloemendaal78** [comment]
Yes, we haven't been able to solve it yet. I will provide the connection profile ASAP.
